SYCL/ESIMD/api/functional/value.hpp

```diff
@@ -135,41 +156,76 @@ template <typename DataT, int NumElems> std::vector<DataT> generate_ref_data() {
     static const DataT nan = value<DataT>::nan();
     static const DataT inf = value<DataT>::inf();
 
-    ref_data.reserve((NumElems > 1) ? NumElems : 6);
-
     // We are using the `double` literals to avoid precision loss for case of
     // the `double` DataT on unexact values like 0.1
-    ref_data.insert(ref_data.end(), {-inf, nan, min, max, -0.0, 0.1});
-    if constexpr (NumElems != 1) {
-      ref_data.insert(ref_data.end(), {-0.1, +0.0});
-      for (size_t i = ref_data.size(); i < NumElems; ++i) {
-        // Store values with exact representation of the fraction part for
-        // every floating point type
-        ref_data.push_back(i + 0.25);
-      }
-    }
+    ref_data = details::construct_ref_data<DataT, NumElems>(
+        0.25, {-inf, nan, min, max, -0.0, 0.1}, {-0.1, +0.0});
   } else if constexpr (std::is_signed_v<DataT>) {
-    ref_data.reserve((NumElems > 1) ? NumElems : 5);
-
-    ref_data.insert(ref_data.end(), {min, min_half, max, max_half, 0});
-    if constexpr (NumElems != 1) {
-      ref_data.insert(ref_data.end(), {min_plus_one, max_minus_one, -1});
-      for (size_t i = ref_data.size(); i < NumElems; ++i) {
-        ref_data.push_back(i);
-      }
-    }
+    ref_data = details::construct_ref_data<DataT, NumElems>(
+        0, {min, min_half, max, max_half, 0},
+        {min_plus_one, max_minus_one, -1});
   } else {
-    // Unsigned integral type
-    ref_data.reserve((NumElems > 1) ? NumElems : 3);
-
-    ref_data.insert(ref_data.end(), {max, max_half, 0});
-    if constexpr (NumElems != 1) {
-      ref_data.insert(ref_data.end(), {max_minus_one});
-      for (size_t i = ref_data.size(); i < NumElems; ++i) {
-        ref_data.push_back(i);
-      }
-    }
+    ref_data = details::construct_ref_data<DataT, NumElems>(
+        0, {max, max_half, 0}, {max_minus_one});
+  }
+  return ref_data;
+}
+
+// Provides std::vector with the reference data according to the obtained data
+// types and number of elements.
+template <typename SrcT, typename DstT, int NumElems>
+std::vector<SrcT> generate_ref_conv_data() {
+  static_assert(std::is_integral_v<SrcT> ||
+                    type_traits::is_sycl_floating_point_v<SrcT>,
+                "Invalid the first data type provided to the "
+                "generate_ref_conv_data function as source type.");
```

Comment:
```suggestion
                "Invalid source type");
```
No need to mention "first" or "second". Also no need to mention the function name. Here and below.

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/696/commits/c9bc321af2b1bf1b9c70b323c48041dd1e79a85e#diff-e031be0968a4e0cad81b5b2879e34920db09777fa3165f4f722fb4e0ab560826R71)